CDCgov/prime-reportstream
# Engagement/jessica/15406 download report cleanup

This PR captures the report not existing error so that we can return a pretty one. Fixes a condition in the pii removal yml. Returns a prettier body for a report

Test Steps:
1. Hit http://localhost:7071/api/reports/download?reportId=220e8b43-7f30-4a8b-ab81-33786d0db5ab with a report id that doesn't exist and make sure it returns a pretty error.
2. Hit http://localhost:7071/api/reports/download?reportId=220e8b43-7f30-4a8b-ab81-33786d0db5ab&removePII=true with a report id that does exist and make sure it does not error out and the body returned is pretty JSON.

## Changes
- Captures the report not existing error so that we can return a pretty one. 
- Fixes a condition in the pii removal yml. 
- Returns a prettier body for a report

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
